bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -1243,73 +1286,223 @@ public void sendNotificationToMobileApp(String customerId, String text, @Nullabl
+                JsonAnnouncementContent content = new JsonAnnouncementContent();
+                if (StringUtils.isEmpty(title)) {
+                    content.display.title = "OpenHAB";
+                } else {
+                    content.display.title = title;
+                }
+                content.display.body = bodyText;
+                content.display.body = speak.replaceAll("<.+?>", "");
+                if (speak.startsWith("<speak>") && speak.endsWith("</speak>")) {
+                    content.speak.type = "ssml";
+                }
+                content.speak.value = speak;
 
-        contentArray[0] = content;
+                contentArray[0] = content;
 
-        parameters.put("content", contentArray);
+                parameters.put("content", contentArray);
 
-        JsonAnnouncementTarget target = new JsonAnnouncementTarget();
-        target.customerId = device.deviceOwnerCustomerId;
-        TargetDevice[] devices = new TargetDevice[1];
-        TargetDevice deviceTarget = new TargetDevice();
-        deviceTarget.deviceSerialNumber = device.serialNumber;
-        deviceTarget.deviceTypeId = device.deviceType;
-        devices[0] = deviceTarget;
-        target.devices = devices;
-        parameters.put("target", target);
+                JsonAnnouncementTarget target = new JsonAnnouncementTarget();
+                target.customerId = devices.get(0).deviceOwnerCustomerId;
+                TargetDevice[] targetDevices = new TargetDevice[devices.size()];
+                for (int i = 0; i < devices.size(); i++) {
+                    Device device = devices.get(i);
+                    TargetDevice deviceTarget = new TargetDevice();
+                    deviceTarget.deviceSerialNumber = device.serialNumber;
+                    deviceTarget.deviceTypeId = device.deviceType;
+                    targetDevices[i] = deviceTarget;
+                }
+                target.devices = targetDevices;
+                parameters.put("target", target);
 
-        String accountCustomerId = this.accountCustomerId;
-        String customerId = StringUtils.isEmpty(accountCustomerId) ? device.deviceOwnerCustomerId : accountCustomerId;
+                String accountCustomerId = this.accountCustomerId;
+                String customerId = StringUtils.isEmpty(accountCustomerId)
+                        ? devices.toArray(new Device[devices.size()])[0].deviceOwnerCustomerId
+                        : accountCustomerId;
 
-        if (customerId != null) {
-            parameters.put("customerId", customerId);
+                if (customerId != null) {
+                    parameters.put("customerId", customerId);
+                }
+                executeSequenceCommandWithVolume(devices.toArray(new Device[devices.size()]), "AlexaAnnouncement",
+                        parameters, ttsVolumes.toArray(new Integer[ttsVolumes.size()]),
+                        standardVolumes.toArray(new Integer[standardVolumes.size()]));
```

Comment:
```suggestion
                executeSequenceCommandWithVolume(devices.toArray(Device[]::new), "AlexaAnnouncement",
                        parameters, ttsVolumes.toArray(Integer[]::new), standardVolumes.toArray(Integer[]::new));
```

Reply:
It seems this doesn't work, got a compilation error.

Reply:
see above